File: Modules/HumanResource/PayrollSettings/WeeklyPayrollEmployee/WeeklyPayrollEmployeeForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.WeeklyPayrollEmployee")]
[BasedOnRow(typeof(WeeklyPayrollEmployeeRow), CheckNames = true)]
public class WeeklyPayrollEmployeeForm
{
    public int WeeklyPayrollSettingId { get; set; }
    public int EmployeeRowId { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public DateTime DeleteDate { get; set; }
    public int DeleteUserId { get; set; }
    public short IsActive { get; set; }
}

File: Modules/ServerTypes/PayrollSettings/WeeklyPayrollEmployeeForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface WeeklyPayrollEmployeeForm {
    WeeklyPayrollSettingId: IntegerEditor;
    EmployeeRowId: IntegerEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
    DeleteDate: DateEditor;
    DeleteUserId: IntegerEditor;
    IsActive: IntegerEditor;
}

export class WeeklyPayrollEmployeeForm extends PrefixedContext {
    static formKey = 'PayrollSettings.WeeklyPayrollEmployee';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!WeeklyPayrollEmployeeForm.init)  {
            WeeklyPayrollEmployeeForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(WeeklyPayrollEmployeeForm, [
                'WeeklyPayrollSettingId', w0,
                'EmployeeRowId', w0,
                'InsertDate', w1,
                'InsertUserId', w0,
                'UpdateDate', w1,
                'UpdateUserId', w0,
                'DeleteDate', w1,
                'DeleteUserId', w0,
                'IsActive', w0
            ]);
        }
    }
}